cpp/core/operators/shuffle/splitter.h

```diff
@@ -25,6 +25,7 @@
 
 #include <random>
```

Comment:
The existing methods for Splitter:

- Split
- Stop
- SpillPartition
- SetCompressType
- SpillFixedSize
- SpillLargestPartition
- CalculateSplitBatchSize

new methods will be added:

- StopForCeleborn
- PushPartition
- Push
- WriteArrowToOutputStream
- PushFixedSize
- PushLargestPartition
- setCelebornClient

Can we try find out if a new abstraction layer of splitter can be leveraged to avoid changing the API too much?

#1077 introduced `SplitterBase` as the base class of Splitter. Probably helpful for this PR?

Reply:
@zhztheplayer  I added a new metric in splitterbase, and put other methods in splitter

Reply:
@zhztheplayer   I plan to put these methods StopForCeleborn, PushFixedSize, GetSplitOptions in SplitterBase, and others in Splitter.